dashboard/app/reporting_email.go

```diff
@@ -1090,12 +1090,24 @@ func ownMailingLists() []string {
 }
 
 var (
-	subjectParser     subjectTitleParser
+	_subjectParser    *subjectTitleParser
+	subjectParserInit sync.Once
 	errAmbiguousTitle = errors.New("ambiguous bug title")
 )
 
+func getSubjectParser(c context.Context) *subjectTitleParser {
+	if getConfig(c) != getConfig(context.Background()) {
+		// For the non-default config, do not cache the parser.
+		return &subjectTitleParser{c: c}
+	}
+	subjectParserInit.Do(func() {
```

Comment:
subjectTitleParser initialization/caching is overly complex. There are 2 levels of lazy initalization with sync.Once. There should be one. Also once we inited subjectTitleParser, it does not need to store context anymore.